p4-16/spec/P4-16-spec.mdk

```diff
@@ -6007,6 +6007,18 @@ Figure [#fig-subparser] shows a diagram of this process.
 Note that since P4 requires declarations to precede uses, it is impossible to
 create recursive (or mutually recursive) parsers.
 
+When a parser is instantiated, all its local declarations of stateful instantiations are evaluated recursively.
+Each instantiation of a parser will have a unique set of local value sets, extern objects, and inner parser instances.
+Thus, invoking a parser instance twice is different from invoking two parser instances each once, where the former
+accesses the same local stateful constructs while the latter access two different copies.
+
+The exactly-once evaluation only applies to local stateful instantiations. For local variable declarations,
+whether in `state`s or out, and whether with initializers or not, they are always evaluated
+when a parser instance is invoked. That is, local variables in a parser never persist across invocations. For variables
+declared outside `state`s, they are evaluated at the beginning of execution.
```

Comment:
Is "evaluated" the right term here?

Reply:
In fact, I think the text could work fine without the second paragraph, both here and below.

Reply:
@MollyDream if you can fix this before the next meeting we should discuss it.